path_role.go

```diff
@@ -243,39 +273,99 @@ func (b *azureAuthBackend) pathRoleCreateUpdate(ctx context.Context, req *logica
 		role = new(azureRole)
 	}
 
-	if policiesRaw, ok := data.GetOk("policies"); ok {
-		role.Policies = policyutil.ParsePolicies(policiesRaw)
-	}
+	if err := role.ParseTokenFields(req, data); err != nil {
+		return logical.ErrorResponse(err.Error()), logical.ErrInvalidRequest
+	}
+
+	// Handle upgrade cases
+	{
+		policiesRaw, ok := data.GetOk("token_policies")
```

Comment:
In all of these cases, the result term (e.g. `policiesRaw` here) isn't used, but the variable name _is_ used in inner blocks. Probably clearer to use `_` for the top level `GetOK`.